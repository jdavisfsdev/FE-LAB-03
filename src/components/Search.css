form {
    display: flex;
    align-items: center;
    justify-content: center;
}
input {
    font-size: 1.2rem;
}

button {
    font-size: 1.2rem;
}

button:hover {
    cursor: pointer;
}