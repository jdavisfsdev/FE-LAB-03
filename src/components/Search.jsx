/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';

export default function Search({ search, onChange, onSubmit }) {
  return (
    <form>
      <input placeholder="Search" id="search" type="text" value={search} name="search" onChange={onChange} onSubmit={onSubmit} /> 
      <button>Search</button> 
    </form>
  );
}

Search.propTypes = {
  search: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func
};
