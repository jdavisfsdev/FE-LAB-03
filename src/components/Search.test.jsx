import React from 'react';
import { render } from '@testing-library/react';
import Search from './Search';

describe('Search component', () => {
  it('renders Article', () => {
    const { asFragment } = render (
      <Search 
        search="defaultValue"
        onSubmit
        onChange
        
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
