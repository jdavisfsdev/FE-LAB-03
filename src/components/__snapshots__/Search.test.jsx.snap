// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Article 1`] = `
<DocumentFragment>
  <form>
    <input
      id="search"
      name="search"
      placeholder="Search"
      type="text"
      value="defaultValue"
    />
    <button>
      Search
    </button>
  </form>
</DocumentFragment>
`;
